src/VatRates/TaxesEuropeDatabaseClient.php

```diff
@@ -85,17 +93,35 @@ private function parseRetrieveVatRateResponse(object $response, string $countryC
      */
     private function call(string $call, array $params): ?object
     {
+        $cacheKey = $this->generateCacheKey($call, $params);
+        if ($this->cache !== null && $this->cache->has($cacheKey)) {
+            return $this->cache->get($cacheKey);
+        }
         try {
             $response = $this->client->__soapCall($call, $params);
-            if (! is_object($response)) {
-                return null;
-            }
-            return $response;
         } catch (SoapFault $fault) {
             throw new VatFetchFailedException(
                 'Could not fetch VAT rate from TEDB: ' . $fault->faultstring,
                 $params
             );
         }
+        if (! is_object($response)) {
+            return null;
+        }
+        if ($this->cache !== null) {
+            $this->cache->set($cacheKey, $response);
```

Comment:
If the cache is configured for multiple components, the ttl here could be exceptionally long.